contracts/adapters/OptimisticAuctionRebalanceExtension.sol

```diff
@@ -0,0 +1,387 @@
+        uint64  liveness;       // The amount of time to dispute proposed transactions before they can be executed.
+        uint256 bondAmount;     // Configured amount of collateral currency to make assertions for proposed transactions.
+        bytes32 identifier;     // Identifier used to request price from the DVM.
+        OptimisticOracleV3Interface optimisticOracleV3;     // Optimistic Oracle V3 contract used to assert proposed transactions.
+    }
+
+    struct ProductSettings{
+        OptimisticRebalanceParams optimisticParams;     // OptimisticRebalanceParams struct containing optimistic rebalance parameters.
+        bytes32 rulesHash;      // IPFS hash of the rules for the product.
+    }
+
+    struct Proposal{
+        bytes32 proposalHash;   // Hash of the proposal.
+        ISetToken product;    // Address of the SetToken to set rules and settings for.
+    }
+
+    /* ============ State Variables ============ */
+    
+    ProductSettings public productSettings; // Mapping of set token to ProductSettings
+    mapping(bytes32 => bytes32) public assertionIds; // Maps proposal hashes to assertionIds.
+    mapping(bytes32 => Proposal) public proposedProduct; // Maps assertionIds to a Proposal.
+
+    // Keys for assertion claim data.
+    bytes public constant PROPOSAL_HASH_KEY = "proposalHash";
+    bytes public constant RULES_KEY = "rulesIPFSHash";
+
+
+    /* ============ Constructor ============ */
+    /*
+      * @dev Initializes the BaseOptimisticAuctionRebalanceExtension with the passed parameters.
+      * 
+      * @param _auctionParams AuctionExtensionParams struct containing the baseManager and auctionModule addresses.
+    */ 
+        constructor(AuctionExtensionParams memory _auctionParams) public AuctionRebalanceExtension(_auctionParams.baseManager, _auctionParams.auctionModule) AssetAllowList(_auctionParams.allowedAssets, _auctionParams.useAssetAllowlist) {
+    
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * ONLY OPERATOR: Add new asset(s) that can be traded to, wrapped to, or claimed
+     *
+     * @param _assets           New asset(s) to add
+     */
+    function addAllowedAssets(address[] memory _assets) external onlyOperator {
+        _addAllowedAssets(_assets);
+    }
+
+    /**
+     * ONLY OPERATOR: Remove asset(s) so that it/they can't be traded to, wrapped to, or claimed
+     *
+     * @param _assets           Asset(s) to remove
+     */
+    function removeAllowedAssets(address[] memory _assets) external onlyOperator {
+        _removeAllowedAssets(_assets);
+    }
+
+    /**
+     * ONLY OPERATOR: Toggle useAssetAllowlist on and off. When false asset allowlist is ignored
+     * when true it is enforced.
+     *
+     * @param _useAssetAllowlist           Bool indicating whether to use asset allow list
+     */
+    function updateUseAssetAllowlist(bool _useAssetAllowlist) external onlyOperator {
+        _updateUseAssetAllowlist(_useAssetAllowlist);
+    }
+
+
```

Comment:
can we add a boolean flag `isOpen` which can be flipped by the `operator` and is checked at the beginning of `proposeRebalance` and `startRebalance` calls?

this is our V1 solution to having a rebalance cadence of every 6 months. we may replace with timelocks in a later version

Reply:
Done. Note that I set it up so that it will be set to "false" every time the startRebalance function has been called, and both propose and startRebalance are gated by this. That means that we will have to flip it bag to "true" for every rebalance. Lmk if that is not what you had in mind.

Reply:
Note that this means to in effect that the extension would be "bricked" if the operator key is lost or set to address(0).

Alternatively I can remove that logic of resetting isOpen to "false" after each rebalance in which case we would have to do that in a manual transaction after each rebalance.